pages/premium.php

```diff
@@ -173,102 +181,81 @@ function premium() {
 			if($player->username_changes > 0){
 				$akCost = 0;
 			}
-			else {
-				$nameCost = 0;
-			}
+            elseif(strtolower($player->user_name) == strtolower($new_name)) {
+                $akCost = 0;
+                $deductNameChanges = 0;
+                $_POST['confirm_name_change'] = 1; //Bypass need to confirm
+            }
+            else {
+                $deductNameChanges = 0;
+            }
 
 			$result = $system->query("SELECT `user_name` FROM `users` WHERE `user_name`='$new_name' LIMIT 1");
 			if($system->db_last_num_rows) {
 				$result = $system->db_fetch();
-				if(strtolower($player->user_name) == strtolower($new_name)) {
-					$nameCost = 0;
-					$akCost = 0;
-				}
-				else if(strtolower($result['user_name']) == strtolower($new_name)) {
-					throw new Exception("Username already in use!");
-				}
+                if(strtolower($result['user_name']) == strtolower($new_name) && $result['user_name'] != $player->user_name) {
+                    throw new Exception("Username already in use!");
+                }
 			}
 
-			if(!isset($_POST['confirm_nameChange'])) {
-				echo "<table class='table'><tr><th>Confirm Change</th></tr>
-				<tr><td style='text-align:center;'>Are you sure you want to change your username?
-				Doing this will also change your log in name to the new selected one.
-				Please remember to clear your inbox before changing name.<br />
-				<form action='$self_link' method='post'>
-				<input type='hidden' name='confirm_nameChange' value='1' />
-				<input type='text' name='new_name' value='{$new_name}'/>
-				<input type='submit' name='name_change' value='Confirm Change' />
-				</form>
-				</td></tr></table>";
-				return true;
-				throw new Exception('');
-			}
+			if(!isset($_POST['confirm_name_change'])) {
+                $confirmation_type = 'confirm_name_change';
+                $confirmation_string = "Are you sure you want to change your username?<br />
+                Doing this will also change your login name to the name you select.<br />
+                Changing your name will make it available for use to anyone!";
+                $additional_form_data = ['new_name' => ['input_type'=>'text', 'value'=>$new_name]];
+                $submit_value = "name_change";
+                $button_value = "Confirm Change";
 
-			$sql = "UPDATE `users` SET `user_name` = '%s',  `premium_credits` = `premium_credits` - %d, `username_changes` = `username_changes` - %d WHERE `user_id` = %d LIMIT 1;";
-			$system->query(sprintf($sql, $new_name, $akCost, $nameCost, $player->user_id));
-			$player->subtractPremiumCredits($akCost, "Username change");
+                require 'templates/premium/purchase_confirmation.php';
```

Comment:
We spend so much effort here declaring variables for the template, I wonder if there's a better way...